packages/utils/dashboard/status-card/states.tsx

```diff
@@ -0,0 +1,119 @@
+import * as React from 'react';
+import { HealthState } from '@openshift-console/dynamic-plugin-sdk';
+import { TFunction } from 'i18next';
+import { InProgressIcon } from '@patternfly/react-icons';
+import {
+  GrayUnknownIcon,
+  BlueSyncIcon,
+  BlueArrowCircleUpIcon,
+  GreenCheckCircleIcon,
+  RedExclamationCircleIcon,
+  YellowExclamationTriangleIcon,
+} from '../../status/icons';
+
+export const healthStateMap = (state: string) => {
+  switch (state) {
+    case '0':
+      return HealthState.OK;
+    case '1':
+      return HealthState.WARNING;
+    case '2':
+      return HealthState.ERROR;
+    default:
+      return HealthState.LOADING;
+  }
+};
+
+export const healthStateMessage = (
+  state: keyof typeof HealthState,
+  t: TFunction
+): string => {
+  switch (state) {
+    case HealthState.OK:
+      return '';
+    case HealthState.UNKNOWN:
+      return t('Unknown');
+    case HealthState.PROGRESS:
+      return t('Pending');
+    case HealthState.UPDATING:
+      return t('Updating');
+    case HealthState.WARNING:
+      return t('Degraded');
+    case HealthState.ERROR:
+      return t('Degraded');
+    case HealthState.LOADING:
+      return t('Loading');
+    case HealthState.UPGRADABLE:
+      return t('Upgrade available');
+    case HealthState.NOT_AVAILABLE:
+      return t('Not available');
+    default:
+      return t('Unknown');
+  }
+};
+
+export const healthStateMapping: {
+  [key in HealthState]: HealthStateMappingValues;
+} = {
+  [HealthState.OK]: {
+    priority: 0,
+    health: HealthState.OK,
+    icon: <GreenCheckCircleIcon title="Healthy" />,
+  },
+  [HealthState.UNKNOWN]: {
+    priority: 1,
+    health: HealthState.UNKNOWN,
+    icon: <GrayUnknownIcon title="Unknown" />,
+  },
+  [HealthState.PROGRESS]: {
+    priority: 2,
+    health: HealthState.PROGRESS,
+    icon: <InProgressIcon title="In progress" />,
+  },
+  [HealthState.UPDATING]: {
+    priority: 3,
+    health: HealthState.UPDATING,
+    icon: <BlueSyncIcon title="Updating" />,
+  },
+  [HealthState.UPGRADABLE]: {
+    priority: 4,
+    health: HealthState.UPGRADABLE,
+    icon: <BlueArrowCircleUpIcon title="Upgrade available" />,
+  },
```

Comment:
we don't need this. but since we are doing this - 

![image](https://user-images.githubusercontent.com/22158580/170042185-e3cc7df6-73b8-448f-a87d-8310acbe931b.png)

and since the health state is imported from SDK, we need to override all of the keys.

`import { HealthState } from '@openshift-console/dynamic-plugin-SDK;`

we can avoid all of these if we have our own copy of the health state.

which way we are going to opt for? @GowthamShanmugam @cloudbehl

Reply:
If you are using this on only one card and if the status which you are going to use is very less, then this is not required. you can use it directly.

If you think this will reduce if condition in your code then trim down only the status which is required and remove the reset.